<!-- seatrac_pkg x110.launch working_mode:=SLAVE gps_en:=TRUE tst_comm_en:=TRUE -->
<!-- seatrac_pkg x110.launch working_mode:=POS_TRACK_PING gps_en:=TRUE tst_comm_en:=TRUE -->
<!-- seatrac_pkg x110.launch working_mode:=POS_TRACK_COMM_PING gps_en:=TRUE tst_comm_en:=TRUE -->
<!-- seatrac_pkg x110.launch working_mode:=POS_TRACK_COMM_NAV gps_en:=TRUE tst_comm_en:=TRUE -->
<launch>
  <!-- Arguments section (must be passed when running it from console) -->
  <arg name="beacon_type"/>
  <arg name="working_mode"/>
  <arg name="gps_en"/>
  <arg name="tst_comm_en"/>
  <!-- Let the console log all the print commands included in the nodes -->
  <env name="ROSCONSOLE_CONFIG_FILE"
       value="$(find seatrac_pkg)/launch/custom_roslogger.conf"/>
  <!-- Beacon node -->
  <node pkg="seatrac_pkg" name="SeaTracBeacon" type="Beacon.py" output="screen">
    <param name="beacon_type" type="string" value="$(arg beacon_type)" />
    <param name="working_mode" type="string" value="$(arg working_mode)"/>
  </node>
  <!-- GPS Node -->
  <group if="$(arg gps_en)">
    <node pkg="seatrac_pkg" name="GpsTracker" type="GpsTracker.py" output="screen"/>
  </group>
  <!-- Test Communication Node -->
  <group if="$(arg tst_comm_en)">
    <node pkg="seatrac_pkg" name="TestNode" type="CustCommTest.py" output="screen">
      <param name="beacon_type" type="string" value="$(arg beacon_type)" />
    </node>
  </group>
</launch>
